joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodReturnTests.scala

```diff
@@ -479,4 +479,35 @@ class MethodReturnTests extends RubyCode2CpgFixture(withDataFlow = true) {
       case None => fail(s"Expected at least one retrun node")
     }
   }
+
+  "Return with methodInvocationWithoutParentheses" in {
+    val cpg = code("""
+        |def foo()
+        | return render json: {}, status: :internal_server_error unless success
+        |end
+        |""".stripMargin)
+
+    inside(cpg.method.name("foo").body.astChildren.isControlStructure.l) {
+      case ifNode :: Nil =>
+        ifNode.controlStructureType shouldBe ControlStructureTypes.IF
+
+        val List(notCall: Call) = ifNode.condition.l: @unchecked
+        notCall.methodFullName shouldBe Operators.logicalNot
+
+        val List(ifReturnTrue: Return) = ifNode.whenTrue.isBlock.astChildren.isReturn.l
+        ifReturnTrue.code shouldBe "return render json: {}, status: :internal_server_error"
+
+        val List(_, jsonArg, statusArg) = ifReturnTrue.astChildren.isCall.argument.l: @unchecked
+        jsonArg.argumentName shouldBe Some("json")
+        jsonArg.code shouldBe "<empty>"
```

Comment:
Could you verify the call node types in this? If `jsonArg` is a call with name `<operator>.hashInitializer` then I'm happy, but if it's a block, then let's add a TODO to handle this differently (like we mentioned offline)